### PR TITLE
docs: add more extension type URLs

### PR DESCRIPTION
**Pull request by kyessenov**
Signed-off-by: Kuat Yessenov <kuat@google.com>

Commit Message: Follow-up to https://github.com/envoyproxy/envoy/pull/21707 with a focus on back-filling more extension type URLs. Renames `extensions_build_config` and `extensions_metadata` to the names in the internal extension registry. For `preserve_case`, we deprecate the short name with the fully qualified name (both names are valid in the interim).

Risk Level: medium, only `preserve_case` name changes
Testing: regression
Docs Changes: yes
Release Notes: yes

### CODE REVIEW

File: source/extensions/http/header_formatters/preserve_case/config.h [hunk 1]
```diff
@@ -38,7 +38,9 @@ class PreserveCaseFormatterFactoryConfig
     : public Envoy::Http::StatefulHeaderKeyFormatterFactoryConfig {
 public:
   // Envoy::Http::StatefulHeaderKeyFormatterFactoryConfig
-  std::string name() const override { return "preserve_case"; }
+  std::string name() const override {
```
Comment (htuch) [inline source/extensions/http/header_formatters/preserve_case/config.h:41]: Is there something that can systematically done to style check these names in presubmit?

File: source/extensions/http/header_formatters/preserve_case/config.h [hunk 2]
```diff
@@ -38,7 +38,9 @@ class PreserveCaseFormatterFactoryConfig
...
+    return "envoy.http.stateful_header_formatters.preserve_case";
```
Comment (htuch) [inline source/extensions/http/header_formatters/preserve_case/config.h:42]: Is this a deprecation or a breaking change?

File: changelogs/current.yaml [hunk 3]
```diff
@@ -116,6 +116,9 @@ minor_behavior_changes:
 - area: logging
   change: |
     changed category name for access log filter extensions to ``envoy.access_loggers.extension_filters``.
+- area: http
```
Comment (htuch) [inline changelogs/current.yaml:119]: Should this be in the `deprecated` section?

**Review by htuch — COMMENTED** [review]
Looks good, couple of questions.

File: source/extensions/http/header_formatters/preserve_case/config.h (hunk 2 above)
Comment (kyessenov) [reply source/extensions/http/header_formatters/preserve_case/config.h:42]: It's not breaking, both names are valid I believe.

File: changelogs/current.yaml (hunk 3 above)
Comment (kyessenov) [reply changelogs/current.yaml:119]: Yes, moving.

File: source/extensions/http/header_formatters/preserve_case/config.h (hunk 1 above)
Comment (htuch) [reply source/extensions/http/header_formatters/preserve_case/config.h:41]: OK, just a suggestion for making this more defensible, I'll merge.

**Review by htuch — APPROVED** [review]

**Review by htuch — APPROVED** [review]

**Pull request merged**